test/watch/test/stdin.test.ts

```diff
@@ -27,6 +27,40 @@ test('rerun current pattern tests', async () => {
   await vitest.waitForStdout('1 passed')
 })
 
+test('rerun failed tests', async () => {
```

Comment:
We'll need a test case that fails in `main`, and is fixed by these changes. This test case does not fail on `main` branch. 

Maybe something like below?

```ts
test('rerun failed tests', async () => {
  const testPath = 'fixtures/error.test.ts'
  const failAssertion = 'expect(1).toBe(2)'
  const testCase = `// Dynamic test case
  import { test, expect } from 'vitest'

  // Failure re-run should run this
  test('failing test', () => {
    ${failAssertion}
  })

  // Failure re-run should NOT run this
  test('passing test', () => {
    expect(1).toBe(1)
  })
  `

  writeFileSync(testPath, testCase, 'utf8')
  cleanups.push(() => rmSync(testPath))

  const vitest = await runVitestCli(...cliArgs)
  await vitest.waitForStdout('Test Files  1 failed | 2 passed (3)')
  await vitest.waitForStdout('Tests  1 failed | 3 passed (4)')

  // Re-run failed tests only
  vitest.write('f')

  // Only a single test should have run
  await vitest.waitForStdout('Test Files  1 failed (1)')

  // Only a single test case should have run
  await vitest.waitForStdout('Tests  1 failed (1)')

  // Fix failing test case
  writeFileSync(testPath, testCase.replace(failAssertion, ''), 'utf8')

  // Only the previously failed test file should run
  await vitest.waitForStdout('Test Files  1 pass (1)')

  // Only the previously failed test case should run
  await vitest.waitForStdout('Tests  1 pass (1)')
})
```